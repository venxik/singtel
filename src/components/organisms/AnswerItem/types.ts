import { TouchableOpacityProps } from '../../atoms';

export type AnswerItemProps = {
  text: string;
} & TouchableOpacityProps;
