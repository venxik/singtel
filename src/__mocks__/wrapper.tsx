import { Provider } from 'react-redux';
import { AppStore, RootState, setupStore } from '../redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';

interface ExtendedRenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function AllTheProviders({ children }: { children: JSX.Element }) {
  return <Provider store={setupStore()}>{children}</Provider>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
