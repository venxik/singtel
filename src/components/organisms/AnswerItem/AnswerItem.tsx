import { StyleSheet, TouchableOpacity } from 'react-native';
import { ComicText, useThemeColor } from '../../atoms';
import { AnswerItemProps } from './types';
import React from 'react';

const AnswerItem = (props: AnswerItemProps) => {
  const { style, lightColor, darkColor, text, disabled, ...otherProps } = props;

  const borderColor = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  return (
    <TouchableOpacity
      style={[
        {
          borderRadius: 6,
          justifyContent: 'center',
          width: 60,
          height: 60,
          borderWidth: 1,
          borderColor,
          margin: 4,
        },
        style,
      ]}
      disabled={disabled}
      {...otherProps}>
      <ComicText testID="answer-item-text" style={styles.text}>
        {text}
      </ComicText>
    </TouchableOpacity>
  );
};

export default React.memo(AnswerItem);

const styles = StyleSheet.create({
  text: { textAlign: 'center', fontSize: 20 },
});
