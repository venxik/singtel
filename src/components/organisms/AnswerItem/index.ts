import AnswerItem from './AnswerItem';

export { AnswerItem };
