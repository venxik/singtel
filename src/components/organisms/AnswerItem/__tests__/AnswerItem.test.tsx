import React from 'react';
import { renderWithProviders } from '../../../../__mocks__';
import { fireEvent } from '@testing-library/react-native';
import AnswerItem from '../AnswerItem';

describe('AnswerItem', () => {
  it(`match snapshot`, () => {
    const tree = renderWithProviders(<AnswerItem text="test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`can be pressed and functional`, () => {
    let number = 1;
    const { getByTestId } = renderWithProviders(
      <AnswerItem testID="answer-item" text="test" onPress={() => (number += 1)} />
    );
    fireEvent.press(getByTestId('answer-item'));
    expect(number).toBe(2);
  });

  it(`can have correct text`, () => {
    const { getByTestId } = renderWithProviders(<AnswerItem testID="answer-item" text="test" />);
    expect(getByTestId('answer-item-text').props.children).toEqual('test');
  });
});
