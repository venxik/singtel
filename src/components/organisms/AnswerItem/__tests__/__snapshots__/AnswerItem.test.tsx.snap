// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerItem match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "#000",
      "borderRadius": 6,
      "borderWidth": 1,
      "height": 60,
      "justifyContent": "center",
      "margin": 4,
      "opacity": 1,
      "width": 60,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "fontSize": 20,
            "textAlign": "center",
          },
          {
            "fontFamily": "comic",
          },
        ],
      ]
    }
    testID="answer-item-text"
  >
    test
  </Text>
</View>
`;

exports[`StyledText match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "#000",
      "borderRadius": 6,
      "borderWidth": 1,
      "height": 60,
      "justifyContent": "center",
      "margin": 4,
      "opacity": 1,
      "width": 60,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "fontSize": 20,
            "textAlign": "center",
          },
          {
            "fontFamily": "comic",
          },
        ],
      ]
    }
    testID="answer-item-text"
  >
    test
  </Text>
</View>
`;
